jupyterlab/commands.py

```diff
@@ -492,8 +508,8 @@ def __init__(self, app_dir, logger=None, kill_event=None, core_config=None):
         )
         self.info = self._get_app_info()
         self.kill_event = kill_event or Event()
-        # TODO: Make this configurable
-        self.registry = 'https://registry.npmjs.org'
+        # TODO: Make this configurable or use the yarn config to set it for methods using _fetch_package_metadata
+        self.registry = registry or 'https://registry.npmjs.org'
```

Comment:
Can we move the default population in here? Is there a particular reason to only apply the registry to some commands?

Reply:
I moved it as a property so subprocess is only called once if `registry` is required.